src/codec.rs

```diff
@@ -1058,6 +1058,18 @@ impl Encode for str {
 	}
 }
 
+impl Decode for Box<str> {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		Ok(String::decode(input)?.into_boxed_str())
```

Comment:
Can you check the already existing `Box` `Decode` implementation. It should be in the same way as this implementation.

Reply:
Do you think that decoded `String` will have some excess capacity or will it have `capacity` == `len`?

Reply:
I've taken a bit different route here

Reply:
should I maybe use `WrapperTypeDecode` like this:

```rs
impl WrapperTypeDecode for Box<str> {
	type Wrapped = String;

	fn decode_wrapped<I: Input>(input: &mut I) -> Result<Self, Error> {
		// Guaranteed to create a Vec with capacity == len
		let vec = Vec::from(Box::<[u8]>::decode(input)?);
		// Guaranteed not to reallocate the vec, only transmute to String
		let str = String::from_utf8(vec).map_err(|_| "Invalid utf8 sequence")?;

		// At this point we have String with capacity == len,
		// therefore String::into_boxed_str will not reallocate
		Ok(str.into_boxed_str())
	}
}
```